core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -1885,7 +1885,7 @@ class LogTest {
     assertTrue("Message payload should be null.", !head.hasValue)
   }
 
-  @Test(expected = classOf[IllegalArgumentException])
+  @Test(expected = classOf[UnexpectedAppendOffsetException])
   def testAppendWithOutOfOrderOffsetsThrowsException() {
```

Comment:
Seems like we want to rename this method and add a test that checks for the true out of order case.